handsontable/src/core.js

```diff
@@ -1520,6 +1504,36 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
     return [[row, propOrCol, value]];
   }
 
+  /**
+   * Process changes prepared for applying to the dataset (unifying list of changes, closing an editor - when needed,
+   * calling a hook).
+   *
+   * @private
+   * @param {Array} changes Array of changes in format `[[row, col, value],...]`.
+   * @param {string} [source] String that identifies how this change will be described in the changes array (useful in afterChange or beforeChange callback). Set to 'edit' if left empty.
+   * @returns {Array} List of changes finally applied to the dataset.
+   */
+  function processChanges(changes, source) {
+    const activeEditor = instance.getActiveEditor();
+    const beforeChangeResult = instance.runHooks('beforeChange', changes, source || 'edit');
```

Comment:
Here you introduced a new feature - previously, the `beforeChange` hook could return only a boolean or undefined value. Right now, there is the possibility to return an array of changes. Is that intentional? If yes, then I don't see the tests that cover that feature, as well there is some missing documentation.

Reply:
Good catch, no it wasn't intentional. Thanks, I've changed a code within 37ac496d6a300dbf969af83484878b26aa5d9c91.